postgresql/installingpostgresqlonibmi.md

```diff
@@ -0,0 +1,171 @@
+# Install and configure Postgresql DB Server on IBM i 
+
+Install postgres yum packages from IBM i ACS Open Source Package Management
+```
+Package list:
+postgresql12
+postgresql12-contrib
+postgresql12-devel
+postgresql12-docs
+postgresql12-libecpg
+postgresql12-libogtypes
+postgresql12-libpq
+postgresql12-server
+```
+
+Create IBM i user: ***POSTGRES*** and give it a password. It's important the user is named POSTGRES because the Postgresql server needs a default user of POSTGRES. At least that was my experience with it. (Also give it *ALLOBJ access for now. You can probably remove later and possibly even disable the IBM i POSTGRES user. ***Not tested.***)
```

Comment:
@abmusse I don't recall us having to do this. If this is true, we should definitely think about trying to create the user with the package or at least documenting it in the package.

Reply:
I found that it is advisable to run PostgreSQL under a separate user account. The user name `postgres` is often used but. https://www.postgresql.org/docs/12/postgres-user.html

Reply:
I also see that Fedora creates a postgres user and group during pre scriptlet: https://src.fedoraproject.org/rpms/postgresql/blob/master/f/postgresql.spec#_885

Reply:
Leaving in place for now. I could not use psql at all unless I was the postgres user during initial set up.

Reply:
Interesting during initial setup who was the user that ran `initdb`?

![image](https://user-images.githubusercontent.com/33973272/83287039-997d7300-a1a6-11ea-87b4-e0cf4b4c7b19.png)

https://www.postgresql.org/docs/current/app-initdb.html

Reply:
postgres user who is already signed in to bash.

Reply:
Gotcha so to use `psql` with other users a new user role needs to be created first.

![image](https://user-images.githubusercontent.com/33973272/83296023-90e06900-a1b5-11ea-80e0-e95d3342a77c.png)


https://www.postgresql.org/docs/8.1/sql-createrole.html

So if `initdb` was run by the `postgres` and another user profile `dbuser` wants to use `psql` a new role will need to be create before access it granted.

Reply:
@kadler It's not required to have the `postgres` but its probably a good idea to create with the rpm package. I'm not sure the `postgres` user should have `*ALLOBJ` authority though.

Reply:
@richardschoen 

I'm fine with creating the `postgres` user but do we really need to give the `postgres` user `*ALLOBJ`? Can we get through with less authorities?